shared-portal-ui/projects/core-ui/src/lib/upload-document/upload-document.component.ts

```diff
@@ -0,0 +1,109 @@
+import { Component } from '@angular/core';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatButton, MatButtonModule } from '@angular/material/button';
+import {
+  MatLabel,
+  MatError,
+  MatFormFieldModule,
+} from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { NgxDropzoneModule } from 'ngx-dropzone';
+import { MatOptionModule } from '@angular/material/core';
+import { MatSelectModule } from '@angular/material/select';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import {MatCheckboxModule} from '@angular/material/checkbox'
+//import { HttpClient } from '@angular/common/http';
+//import { ApiConfiguration } from '../shared/api/api-configuration';
+
+@Component({
+  selector: 'app-upload-document',
+  standalone: true,
+  imports: [
+    MatButtonModule,
+    ReactiveFormsModule,
+    FormsModule,
+    MatFormFieldModule,
+    MatCheckboxModule,
+    MatLabel,
+    MatSelectModule,
+    MatOptionModule,
+    MatError,
+    NgxDropzoneModule,
+    MatIconModule,
+  ],
+  templateUrl: './upload-document.component.html',
+  styleUrl: './upload-document.component.scss',
+})
+export class UploadDocumentComponent {
+  fileToUpload: File | null = null;
+
+  constructor(private _snackBar: MatSnackBar) {}
+  public files: any[] = [];
+
+  onSelect(event: any) {
+    this.fileToUpload = event.addedFiles[0];
+  }
+
+  onRemove() {
+    this.fileToUpload = null;
+  }
+
+  deleteFile(f: any) {
+    this.files = this.files.filter(function (w) {
+      return w.name != f.name;
+    });
+    this._snackBar.open('Successfully delete!', 'Close', {
+      horizontalPosition: 'center',
+      verticalPosition: 'top',
+      duration: 5000,
+    });
+  }
+
+  showUpload = false;
+
+  openUploadFile() {
+    this.showUpload = true;
+  }
+
+  closeUploadFile() {
+    this.showUpload = false;
+  }
+
+  handleFileInput(event: any) {
+    this.fileToUpload = event.target.files[0];
+  }
+  isFileUploading = false;
+
+  fileUpload() {
```

Comment:
in a future PR will this be something the parent component can subscribe to and run code similar to the commented out code?